### PR TITLE
Update design for uneditable descriptions

### PR DESCRIPTION
**Pull request by ttannis**
### Summary of Changes

Our updated design guidance for non-editable metadata involves leveraging the same UI for editable metadata, however For the un-editable case the edit icon has different color styling on hover, shows some help text, and clicking on the item will direct the user to the external tool where they can edit the information.

This PR does the following:
1. Updates the capability of `EditableSection` to support these features.
2. Updates the description on `DashboardPage` to use this new UI.
3. Moves `EditableSection` to  the `common` directory and updates imports accordingly.
4. Modifies some styles accordingly.

### Tests

Tests updated to cover changes

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/common/EditableSection/index.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        isEditing: this.state.isEditing,
+        setEditMode: this.setEditMode,
+      });
+    }) : this.props.children;
+
+    return (
+      <section className="editable-section">
+        <div className="section-title title-3">
+          { EditableSection.convertText(title) }
+          { !readOnly ? this.renderButton() : this.renderReadOnlyButton() }
```
Comment (ttannis) [inline amundsen_application/static/js/components/common/EditableSection/index.tsx:85]: This file was moved so there is no clear diff, but compare against https://github.com/lyft/amundsenfrontendlibrary/blob/master/amundsen_application/static/js/components/TableDetail/EditableSection/index.tsx

**Comment by ttannis** [comment]
@Golodhros for review.

As a follow-up, any experience with getting `enzyme` to play well with verifying interactions with Bootstrap `OverlayTrigger` or `Popover`. At most I'd add another assertion that simulates the interaction and verifies the desired text is rendered, but I wasn't able to get that working so far.

File: amundsen_application/static/js/components/DashboardPage/index.tsx [hunk 2]
```diff
@@ -172,23 +173,29 @@ export class DashboardPage extends React.Component<DashboardPageProps, Dashboard
...
+                hasDescription &&
+                <div>
+                  { dashboard.description }
+                </div>
+              }
+              {
+                !hasDescription &&
+                <a
+                 className="edit-link body-2"
+                 target="_blank"
```
Comment (Golodhros) [inline amundsen_application/static/js/components/DashboardPage/index.tsx:192]: I found it is a good practice to add the rel=”noopener noreferrer” attribute to all links that have `target="_blank"`. 

Maybe it is not strictly applicable to this case in particular, but either way is good to just add it always. Check [more info here](https://web.dev/external-anchors-use-rel-noopener/).

File: amundsen_application/static/js/components/common/EditableSection/index.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+export class EditableSection extends React.Component<EditableSectionProps, EditableSectionState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isEditing: false,
+    }
+  }
```
Comment (Golodhros) [inline amundsen_application/static/js/components/common/EditableSection/index.tsx:31]: Nit: It depends a bit on the babel configuration of the project, but I usually found more readable and easy to set the state as an instance variable outside the constructor, like this:
```
export class EditableSection extends ... {
  state = {isEditing: false};
...
```
This also avoids the common mistake of forgetting to call 'super'!

Comment (ttannis) [reply amundsen_application/static/js/components/common/EditableSection/index.tsx:31]: Thanks for the comment, I hadn't seen this yet. I'll read about it more to see if there's other benefits.

**Pull request merged**